build fail real:
```
======================================================================
FAIL: test_product_fixture (corehq.apps.commtrack.tests.test_fixture.FixtureTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/commtrack/tests/test_fixture.py", line 115, in test_product_fixture
    ElementTree.tostring(fixture[0])
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/app_manager/tests/util.py", line 28, in assertXmlEqual
    _check_shared(expected, actual, LXMLOutputChecker(), "xml")
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/app_manager/tests/util.py", line 82, in _check_shared
    raise AssertionError(message)
AssertionError: XML mismatch
--- want.xml
+++ got.xml
@@ -1,5 +1,17 @@
 <fixture id="commtrack:products" user_id="66d7a7d5b5b58d41f54dcafedac170ac">
   <products>
+    <product id="f140eb18c372cc20212098e2817486aa">
+      <name>fubrsvdmzzpxtnzduslk</name>
+      <unit>egjrhthzkbylwlwgvpmc</unit>
+      <code>vwqwdjpxtsykpcngimrt</code>
+      <description>lqxqgvkffxapsyswkmkf</description>
+      <category>hwzlwaurccpvltfssbja</category>
+      <program_id>vawhdwsufkklzrvecxex</program_id>
+      <cost>0</cost>
+      <product_data>
+        <special_number>555111</special_number>
+      </product_data>
+    </product>
     <product id="f140eb18c372cc20212098e281748651">
       <name>hlqtprxuaqbxyhavlmqy</name>
       <unit>hwzogcnetyaebypkcoqf</unit>
@@ -18,17 +30,5 @@
       <program_id>fsirlsfwuzfijfuxvlod</program_id>
       <cost>72.2562</cost>
     </product>
-    <product id="f140eb18c372cc20212098e2817486aa">
-      <name>fubrsvdmzzpxtnzduslk</name>
-      <unit>egjrhthzkbylwlwgvpmc</unit>
-      <code>vwqwdjpxtsykpcngimrt</code>
-      <description>lqxqgvkffxapsyswkmkf</description>
-      <category>hwzlwaurccpvltfssbja</category>
-      <program_id>vawhdwsufkklzrvecxex</program_id>
-      <cost>0</cost>
-      <product_data>
-        <special_number>555111</special_number>
-      </product_data>
-    </product>
   </products>
 </fixture>
----------------------------------------------------------------------
```

--- a/corehq/apps/commtrack/tests/test_fixture.py
+++ b/corehq/apps/commtrack/tests/test_fixture.py
@@ -29,7 +29,7 @@ class FixtureTest(CommTrackTest, TestXmlMixin):
         self.product_names = get_product_name()
 
     def generate_product_xml(self, user, randomize_data=True):
-        products = ''
+        products = []
         product_list = Product.by_domain(user.domain)
         self._initialize_product_names(len(product_list))
         for i, product in enumerate(product_list):
@@ -58,7 +58,7 @@ class FixtureTest(CommTrackTest, TestXmlMixin):
                 product_data = {}
                 custom_data_xml = ''
 
-            products += '''
+            product_xml = '''
                 <product id="{id}">
                     <name>{name}</name>
                     <unit>{unit}</unit>
@@ -91,7 +91,10 @@ class FixtureTest(CommTrackTest, TestXmlMixin):
             product.product_data = product_data
             product.save()
 
-        return products
+            products.append((product, product_xml))
+
+        products.sort(key=lambda p: p[0].code)
+        return ''.join(product_xml for product, product_xml in products)
 
     def generate_product_fixture_xml(self, user, randomize_data=True):
         return """

